### PR TITLE
Feature/rtt 12 filter list view frontend

### PR DESCRIPTION
**Pull request by kiddhustle**
Adds UI changes for the search fillters full details can be found here: https://uktrade.atlassian.net/browse/RTT-12

PLEASE NOTE: the actual search functionality backend is part of a separate PR, so the actual search fillters are not expected to work here

### CODE REVIEW

File: app/enquiries/templates/partials/enquiry_filters.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+             {{owner.user.first_name}} {{owner.user.last_name}}
+          </label>
+        </div>
+        {% endfor %}
+      </div>
+    </fieldset>
+  </div>
+    <div class="govuk-form-group">
+    <label class="govuk-label" for="date_created_before">
+      Date received before
```
Comment (mforner13) [inline app/enquiries/templates/partials/enquiry_filters.html]: This didn't read very clearly for me - would 'Received before' and 'Received after' make more sense? it's clear it's a date from the dd/mm/yyyy

File: app/enquiries/templates/partials/enquiry_filters.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    </label>
+    <input class="govuk-input" id="date_created_before" name="date_created_before" type="date" value="{{query_params.date_created_before}}">
+  </div>
+  <div class="govuk-form-group">
+    <label class="govuk-label" for="date_created_after">
+      Date received after
```
Comment (mforner13) [inline app/enquiries/templates/partials/enquiry_filters.html]: Same as above - 'Received after'

File: app/enquiries/templates/partials/enquiry_filters.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  </div>
+  <div class="govuk-form-group">
+    <label class="govuk-label" for="enquirer_email">
+      Enquirer email address
+    </label>
+    <input class="govuk-input" id="enquirer_email" name="enquirer_email" type="search" value="{{query_params.enquirer_email}}">
+  </div>
+  <div class="govuk-form-group">
+    <label class="govuk-label" for="date_added_to_datahub__lt">
+      Date added to Data Hub before
```
Comment (mforner13) [inline app/enquiries/templates/partials/enquiry_filters.html]: Is this when the company was added to Data Hub? If so, I think 'Company added to Data Hub before' would be clearer. Also if this is the case, what's the need for this filter?

File: app/enquiries/templates/partials/enquiry_filters.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    </label>
+    <input class="govuk-input" id="date_added_to_datahub__lt" name="date_added_to_datahub_before" type="date" value="{{query_params.date_added_to_datahub_before}}">
+  </div>
+  <div class="govuk-form-group">
+    <label class="govuk-label" for="date_added_to_datahub__gt">
+      Date added to Data Hub after
```
Comment (mforner13) [inline app/enquiries/templates/partials/enquiry_filters.html]: ^^^

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 1 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: @rahulbist what do you reckon regarding the wording?

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 3 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: again check wit @rahulbist regarding wording

**Comment by kiddhustle** [comment]
> Looks good, let me know what you think of the suggestions :)
> Will be good to test the functionality with the back end if possible?

@mforner13 I think these copy commenst are more for @rahulbist I think.

He will have more insight into the what will make sense to users. I have asked him provide some feedback

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 1 above)
Comment (rahulbist) [reply app/enquiries/templates/partials/enquiry_filters.html]: Happy for changes to be made to rectify @mfroner13 comments.

Please amend the following:

Date received before to 'Received before'
Date received after to 'Received after'

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 3 above)
Comment (rahulbist) [reply app/enquiries/templates/partials/enquiry_filters.html]: Happy for changes to be made to rectify @mfroner13 comments.

Please amend the following:

Date added to Data Hub before to 'Company added to Data Hub before'
Date added to Data Hub after to 'Company added to Data Hub after'

The need for these two filters is to allow IST to obtain enquiries they have uploaded to Data Hub during a certain period to gather information for the Marketing team (this is something they do to obtain information for their weekly and monthly marketing reports)

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 2 above)
Comment (rahulbist) [reply app/enquiries/templates/partials/enquiry_filters.html]: Happy for changes to be made to rectify @mforner13 comments.

Please amend the following:

Date received after to 'Received after'

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 4 above)
Comment (rahulbist) [reply app/enquiries/templates/partials/enquiry_filters.html]: Happy for changes to be made to rectify @mforner13 comments.

Please amend the following:

Date added to Data Hub after to 'Company added to Data Hub after'

The need for this filters is to allow IST to obtain enquiries they have uploaded to Data Hub during a certain period to gather information for the Marketing team (this is something they do to obtain information for their weekly and monthly marketing reports)

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 1 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: amended

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 3 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: amended

File: app/enquiries/templates/snippets/enquiry_item.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                        href="{% url 'enquiry-detail' obj.id %}">{{ enquiry.company_name }}</a>
+                </h3>
+            </div>
+            <div class="entity-badges">
+                <div class="entity-badges-list">
+                    <div class="entity-badges-item">
+                        <span>{{ enquiry.enquiry_stage }}</span>
+                    </div>
+                    <div class="entity-badges-item">
+                        <span>{% if enquiry.owner %}{{ enquiry.owner.user }} {%else%} {{ "Unassigned" }}
```
Comment (mcbhenwood) [inline app/enquiries/templates/snippets/enquiry_item.html]: Is there a test for the "Unassigned" value?

File: app/enquiries/templates/snippets/enquiry_field_checkbox.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      </span>
+      <div class="govuk-checkboxes govuk-checkboxes--small">
+        {% for value, label in field.choices %}
+        <div class="govuk-checkboxes__item">
+            <input class="govuk-checkboxes__input" id="{{ field.name }}_{{value}}" name="{{ field.name }}" type="checkbox" value="{{value}}" {% query_params_value_selected value 'enquiry_stage' query_params ' checked'  %}>
+            <label class="govuk-label govuk-checkboxes__label" for="{{ field.name }}_{{value}}">
+              {{label}}
+            </label>
+        </div>
+        {% endfor %}
```
Comment (mcbhenwood) [inline app/enquiries/templates/snippets/enquiry_field_checkbox.html]: Where do we test that `field.choices` is rendered as expected into the checkboxes?

File: app/enquiries/tests/test_filters.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+faker = Faker(["en_GB", "en_US", "ja_JP"])
+
+
+class EnquiryViewFiltersTestCase(TestCase):
+    def setUp(self):
+        self.faker = Faker()
+        self.client = Client()
+
+    def test_enquiry_filters_render(self):
```
Comment (mcbhenwood) [inline app/enquiries/tests/test_filters.py]: Besides testing that the filter controls render, we'd also need to see tests which verify that the filters achieve the expected results.

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 2 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: amended

File: app/enquiries/templates/partials/enquiry_filters.html (hunk 4 above)
Comment (kiddhustle) [reply app/enquiries/templates/partials/enquiry_filters.html]: amended

File: app/enquiries/templates/snippets/enquiry_field_checkbox.html (hunk 6 above)
Comment (kiddhustle) [reply app/enquiries/templates/snippets/enquiry_field_checkbox.html]: @mcbhenwood I have added tests to check that the checkbox values and labels are rendered correctly.

File: app/enquiries/tests/test_filters.py (hunk 7 above)
Comment (kiddhustle) [reply app/enquiries/tests/test_filters.py]: @mcbhenwood I'm not sure how to how to test the functionlity within the scope of this scope (frontend only). Maybe one the backend has been merged in we could write some browser tests?

**Pull request merged**